### ISSUE TITLE
[Issue] ags not running 

### ISSUE THREAD

**Issue by NextStep-IM** [end-4/dots-hyprland#350]
- **(For usage) I have read the [wiki](https://end-4.github.io/dots-hyprland-wiki/en/i-i/02usage)**
- **(For troubleshooting) I have read the [wiki](https://end-4.github.io/dots-hyprland-wiki/en/i-i/03troubleshooting)**
- Linux distro: Arco Linux
---
## The issue
Ags just stopped running when I updated to the latest commits.
<details>
<summary>Click to expand</summary>

```plain

(com.github.Aylur.ags:379774): Gjs-WARNING **: 15:47:50.142: JS ERROR: SyntaxError: JSON.parse: unexpected end of data at line 1 column 1 of the JSON data @ file:///home/samtheradiant/.config/ags/services/gemini.js:219:18
appendHistory@file:///home/samtheradiant/.config/ags/services/gemini.js:219:18
loadHistory@file:///home/samtheradiant/.config/ags/services/gemini.js:212:14
GeminiService@file:///home/samtheradiant/.config/ags/services/gemini.js:163:38
@file:///home/samtheradiant/.config/ags/services/gemini.js:328:16
_init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:266:34



(com.github.Aylur.ags:379774): Gtk-CRITICAL **: 15:47:50.212: gtk_widget_set_accel_path: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed

(com.github.Aylur.ags:379774): Gtk-CRITICAL **: 15:47:50.212: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed

(com.github.Aylur.ags:379774): Gtk-CRITICAL **: 15:47:50.213: gtk_widget_set_accel_path: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed

(com.github.Aylur.ags:379774): Gtk-CRITICAL **: 15:47:50.213: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed

(com.github.Aylur.ags:379774): Gtk-CRITICAL **: 15:47:50.213: gtk_widget_set_accel_path: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed

(com.github.Aylur.ags:379774): Gtk-CRITICAL **: 15:47:50.213: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed

(com.github.Aylur.ags:379774): Gjs-Console-CRITICAL **: 15:47:50.224: Error: There is no window named overview
getWindow@resource:///com/github/Aylur/ags/app.js:140:27
update@file:///home/samtheradiant/.config/ags/modules/overview/overview_hyprland.js:334:26
default/OverviewRow/setup/<@file:///home/samtheradiant/.config/ags/modules/overview/overview_hyprland.js:377:60
hook/<@resource:///com/github/Aylur/ags/widgets/widget.js:43:25
_init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:266:34
  

(com.github.Aylur.ags:379774): Gjs-CRITICAL **: 15:47:50.224: JS ERROR: TypeError: (intermediate value).getWindow(...) is undefined
update@file:///home/samtheradiant/.config/ags/modules/overview/overview_hyprland.js:334:26
default/OverviewRow/setup/<@file:///home/samtheradiant/.config/ags/modules/overview/overview_hyprland.js:377:60
hook/<@resource:///com/github/Aylur/ags/widgets/widget.js:43:25
_init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:266:34


(com.github.Aylur.ags:379774): Gjs-Console-CRITICAL **: 15:47:50.225: Error: There is no window named overview
getWindow@resource:///com/github/Aylur/ags/app.js:140:27
update@file:///home/samtheradiant/.config/ags/modules/overview/overview_hyprland.js:334:26
default/OverviewRow/setup/<@file:///home/samtheradiant/.config/ags/modules/overview/overview_hyprland.js:398:39
hook/<@resource:///com/github/Aylur/ags/widgets/widget.js:43:25
_init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:266:34
  

(com.github.Aylur.ags:379774): Gjs-CRITICAL **: 15:47:50.225: JS ERROR: TypeError: (intermediate value).getWindow(...) is undefined
update@file:///home/samtheradiant/.config/ags/modules/overview/overview_hyprland.js:334:26
default/OverviewRow/setup/<@file:///home/samtheradiant/.config/ags/modules/overview/overview_hyprland.js:398:39
hook/<@resource:///com/github/Aylur/ags/widgets/widget.js:43:25
_init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:266:34


(com.github.Aylur.ags:379774): Gjs-Console-CRITICAL **: 15:47:50.225: Error: There is no window named overview
getWindow@resource:///com/github/Aylur/ags/app.js:140:27
update@file:///home/samtheradiant/.config/ags/modules/overview/overview_hyprland.js:334:26
default/OverviewRow/setup/<@file:///home/samtheradiant/.config/ags/modules/overview/overview_hyprland.js:377:60
hook/<@resource:///com/github/Aylur/ags/widgets/widget.js:43:25
_init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:266:34
  

(com.github.Aylur.ags:379774): Gjs-CRITICAL **: 15:47:50.225: JS ERROR: TypeError: (intermediate value).getWindow(...) is undefined
update@file:///home/samtheradiant/.config/ags/modules/overview/overview_hyprland.js:334:26
default/OverviewRow/setup/<@file:///home/samtheradiant/.config/ags/modules/overview/overview_hyprland.js:377:60
hook/<@resource:///com/github/Aylur/ags/widgets/widget.js:43:25
_init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:266:34


(com.github.Aylur.ags:379774): Gjs-Console-CRITICAL **: 15:47:50.226: Error: There is no window named overview
getWindow@resource:///com/github/Aylur/ags/app.js:140:27
update@file:///home/samtheradiant/.config/ags/modules/overview/overview_hyprland.js:334:26
default/OverviewRow/setup/<@file:///home/samtheradiant/.config/ags/modules/overview/overview_hyprland.js:398:39
hook/<@resource:///com/github/Aylur/ags/widgets/widget.js:43:25
_init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:266:34
  

(com.github.Aylur.ags:379774): Gjs-CRITICAL **: 15:47:50.226: JS ERROR: TypeError: (intermediate value).getWindow(...) is undefined
update@file:///home/samtheradiant/.config/ags/modules/overview/overview_hyprland.js:334:26
default/OverviewRow/setup/<@file:///home/samtheradiant/.config/ags/modules/overview/overview_hyprland.js:398:39
hook/<@resource:///com/github/Aylur/ags/widgets/widget.js:43:25
_init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:266:34
```

</details>

**Comment by NotAF0e** [comment]
I have a similar issue. I reloaded AGS with the key-bind but then all widgets disappeared and i was left with only a wallpaper and my cursor. I switched to another user or instance (not sure what it is called) using control+alt+f2 to close hyprland. I also rebooted many times but the widgets would never reappear. This may be an AGS issue but I am not certain.

**Comment by Euroysx** [comment]
on terminal run ags -q  and ags -Init

**Comment by howjack** [comment]
For me it worked by running `ags -q`, `ags --clear-cache` and `ags -init`

**Comment by cristoxdxd** [comment]
I tried the command `ags --clear-cache` after the `ags -q` command and get this:

`(com.github.Aylur.ags:2559): GLib-GIO-CRITICAL **: 17:51:57.449: g_application_run: assertion '!application->priv->must_quit_now' failed`

**Comment by NextStep-IM** [comment]
> For me it worked by running `ags -q`, `ags --clear-cache` and `ags --init`

Worked for me too.



> I tried the command `ags --clear-cache` after the `ags -q` command and get this:
> 
> `(com.github.Aylur.ags:2559): GLib-GIO-CRITICAL **: 17:51:57.449: g_application_run: assertion '!application->priv->must_quit_now' failed`

Same error here.

**Comment by Euroysx** [comment]
ags --help
See the menu, try to stop ags by force 
And try ags -I

And next time dont reload bya keybind instead the fix

**Comment by clsty** [comment]
It seems that the reason comes from an empty `~/.cache/ags/user/ai/chats/gemini.txt`.

This is a workaround:
```bash
rm ~/.cache/ags/user/ai/chats/gemini.txt
```

> For me it worked by running `ags -q`, `ags --clear-cache` and `ags --init`

I guess this does the same thing, but more than needed.

**Comment by clsty** [comment]
The main problem comes from the `saveHistory()` of `.config/ags/services/gemini.js` as it creates that empty `~/.cache/ags/user/ai/chats/gemini.txt` every time when ags runs, which cause the `appendHistory()` JSON parse error.

**Comment by Euroysx** [comment]
And the volume controler - and + doesn't work too.

**Comment by clsty** [comment]
> And the volume controler - and + doesn't work too.

@Euroysx Please open another issue for this problem.

**Comment by cristoxdxd** [comment]
> It seems that the reason comes from an empty `~/.cache/ags/user/ai/chats/gemini.txt`.
> ```shell
> rm ~/.cache/ags/user/ai/chats/gemini.txt
> pkill ags;ags
> ```

So, I need to re-run this every time I turn on my computer
Any idea to avoid this, please

**Comment by NextStep-IM** [comment]
@cristoxdxd I believe the issue has been resolved with 883f388
